PINCache/PINCache.m

```diff
@@ -74,6 +74,22 @@ + (instancetype)sharedCache
 
 #pragma mark - Public Asynchronous Methods -
 
+- (void)containsObjectForKey:(NSString *)key block:(PINCacheObjectContainmentBlock)block
+{
+    if (!key || !block) {
```

Comment:
If you look into the header file you can see that we declare key and block as `nonnull`. So currently the contract explicitly requires a key and block and the API consumer should not pass nil in one of the two arguments. We could discuss changing the block to `nullable`, but as the method does not have a side effect like calling `setObject:forKey:block:` it makes no really sense to call the method without a key or block argument. I decided to define both arguments `nonnull`. In this case we can early return if no key or block is given to prevent bad behavior if called from within Obj-C, Swift should already prevent API consumer to call it with a nil parameter in the first place.